tests/test_renepay.py

```diff
@@ -624,3 +624,60 @@ def test_fees(node_factory):
     source.rpc.call("renepay", {"invstring": invstr})
     invoice = only_one(dest.rpc.listinvoices("inv2")["invoices"])
     assert invoice["amount_received_msat"] == Millisatoshi("150000sat")
+
+
+def test_local_htlcmax0(node_factory):
+    """Testing a simple pay route when local channels have htlcmax=0."""
+    l1, l2, l3 = node_factory.line_graph(3, wait_for_announce=True)
+    l1.rpc.setchannel(l2.info["id"], htlcmax=0)
+    inv = l3.rpc.invoice(123000, "test_renepay", "description")["bolt11"]
+    details = l1.rpc.call("renepay", {"invstring": inv})
+    assert details["status"] == "complete"
+    assert details["amount_msat"] == Millisatoshi(123000)
+    assert details["destination"] == l3.info["id"]
+
+
+def test_htlcmax0(node_factory):
+    """
+    Topology:
+    1----2----4
+    |         |
+    3----5----6
+    Tests the plugin when some routes have htlc_max=0.
+    """
+    opts = [
+        {"disable-mpp": None, "fee-base": 0, "fee-per-satoshi": 0},
+        {"disable-mpp": None, "fee-base": 0, "fee-per-satoshi": 0},
+        {"disable-mpp": None, "fee-base": 0, "fee-per-satoshi": 0},
+        {
+            "disable-mpp": None,
+            "fee-base": 0,
+            "fee-per-satoshi": 0,
+            "htlc-maximum-msat": 0,
+        },
+        {
+            "disable-mpp": None,
+            "fee-base": 0,
+            "fee-per-satoshi": 0,
+            "htlc-maximum-msat": 800000000,
+        },
+        {"disable-mpp": None, "fee-base": 0, "fee-per-satoshi": 0},
+    ]
+    l1, l2, l3, l4, l5, l6 = node_factory.get_nodes(6, opts=opts)
+    start_channels(
+        [
+            (l1, l2, 10000000),
+            (l2, l4, 1000000),
+            (l4, l6, 1000000),
+            (l1, l3, 10000000),
+            (l3, l5, 1000000),
+            (l5, l6, 1000000),
+        ]
+    )
+
+    inv = l6.rpc.invoice("600000sat", "inv", "description")
+
+    l1.rpc.call("renepay", {"invstring": inv["bolt11"]})
+    l1.wait_for_htlcs()
+    invoice = only_one(l6.rpc.listinvoices("inv")["invoices"])
+    assert invoice["amount_received_msat"] >= Millisatoshi("600000sat")
```

Comment:
This is not relevant to this PR, but I think that maybe would be good for this kind of testing to know what is the path that the payment is taking.

Maybe we can add this as a `pay` result?

Reply:
Neither `pay` does this, we get this kind of information from logs alone.

Reply:
>Neither pay does this, we get this kind of information from logs alone.

My idea was also to implement it inside vanilla pay

Reply:
It would be nice if renepay gave the successful paths in the response, though that's pretty verbose!

Reply:
What if the response from `waitsendpay` (as well as the payment success/failure notification) had a field with the route?
In `renepay` there is an internal database of these routes pushed to `sendpay` so that the plugin knows when a specific `(groupid, partid)` fail or succeeds it can recover which route was that and act accordingly, for example by updating the uncertainty/knowledge. If the route was already provided through the result of `waitsendpay` it would be easier and less fragile on `renepay`'s side.